engine/title.asm

```diff
@@ -3,17 +3,654 @@ INCLUDE "constants.asm"
 SECTION "Title screen", ROMX[$5D8C], BANK[$01]
 
 IntroSequence:: ; 5d8c
-    ; TODO
+	ld hl, GameFreakLogo ;E4000
+	ld a, $39
+	call FarCall_hl
+	jr c, TitleSequenceStart ;5dae
+	ld a, [wOpeningType] ;cc38
+	and a
+	jr z, .opening_sequence_start
+	
+.pikachu_minigame_start
+	ld hl, PikachuMiniGame ;E0000
+	ld a, $38
+	call FarCall_hl
+	jr TitleSequenceStart ;5dae
 
+.opening_sequence_start
+	ld hl, OpeningSequence ;E432F
+	ld a, $39
+	call FarCall_hl
+
+TitleSequenceStart::
+	call TitleSequenceInit
+	ld hl, SetTitleBGDecorationBorder ;91D2
+	ld a, $02
+	call FarCall_hl
+	
+.loop
+	call TitleScreenMain ;5fb8
+	jr nc, .loop
+	
+	call ClearPalettesAndWait ;361e -clear palettes and wait
+	call ClearSprites
+	ld a, $01
+	ldh [$FFDE], a
+	call ClearTileMap
+	call ResetPalette ;33a
+	
+	ld a, [wSequence + 1]	; wSequence + 1 = $CB5F
+	ld e, a
+	ld d, $00
+	ld hl, TitleScreenJumpTable
+	add hl, de
+	add hl, de
+	ldi a, [hl]
+	ld h, [hl]
+	ld l, a
+	
+	jp hl
+
+TitleScreenJumpTable::
+	dw MainMenu
+	dw DebugMenu ;4031
+	dw SRAMClearMenu ;61c6
+	dw IntroSequence
+
+TitleSequenceInit::
+	call ClearPalettes
+	
+	xor a
+	ldh [$FFE8], a
+	ldh [$FFDA], a
+	ldh [$FFD9], a
+	
+	ld de, $0000 ;Stop the music.
+	call PlayMusic
+	
+	call ClearTileMap
+	call DisableLCD
+	call ClearSprites
+	
+	ld a, $23
+	ld hl, InitEffectObject	; 8CCFD - create another object?
+	call FarCall_hl
+	ld hl, $8000
+	ld bc, $1800
+
+.clear_loop
+	ld [hl], $00
+	inc hl
+	dec bc
+	ld a, b
+	or c
+	jr nz, .clear_loop
+
+	ld hl, TitleScreenGFX	; TitleScreenGFX = $47CF
+	ld de, $9410
+	ld bc, $00D0
+	ld a, $04
+	call FarCopyData
+	
+	ld hl, TitleScreenVersionGFX ;1095F
+	ld de, $9600
+	ld bc, $0180
+	ld a, $04
+	call FarCopyData
+	
+	ld hl, TitleScreenHoOhGFX	;10ADF
+	ld de, $9000
+	ld bc, $0310
+	ld a, $04
+	call FarCopyData
+	
+	ld hl, TitleScreenLogoGFX ;10DEF
+	ld de, $8800
+	ld bc, $03A0
+	ld a, $04
+	call FarCopyData
+	
+	ld hl, TitleScreenGoldLogoGFX	;1118F
+	ld de, $8BA0
+	ld bc, $0140
+	ld a, $04
+	call FarCopyData
+	
+	call SetTitleArt ;6288
+	ld hl, wTileMapBackup	; wTileMapBackup = $C408
+	ld a, $24
+	ldi [hl], a
+	ld a, $00
+	ldi [hl], a
+	
+	ld hl, $9800
+	ld bc, $0800
+	ld a, $7F
+	call ByteFill
+	
+	ld b, $06
+	call GetSGBLayout
+	call EnableLCD
+	ld a, $01	
+	ldh [$FFDE], a
+	call WaitBGMap
+	xor a
+	ldh [$FFDE], a
+	ld hl, wSequence	; wSequence = $CB5E
+	ldi [hl], a
+	ldi [hl], a
+	ldi [hl], a
+	ld [hl], a
+	
+	call .load_position_table
+		
+	
+	ld a, $1A
+	ldh [rBGP], a
+	ld a, $E4
+	ldh [rOBP0], a
+	ret
+	
+.load_position_table:
+	ld hl, FirePositionTable
+	ld c, $06	;load 6 flying objects on the screen
+	
+.set_fire_note_loop
+	push bc
+	ld e, [hl]
+	inc hl
+	ld d, [hl]
+	inc hl
+	push hl
+	ld a, $2E	;title fire/note object effect type?
+	call AddEffectObject ;3CA8 add additional object?
```

Comment:
This is `_InitSpriteAnimStruct` (sic) in pokecrystal (this is (sic) because naming convention would have the ROMX function have the `_` prefix, while the home function which calls the ROMX function would not have a `_` prefix).

Reply:
Done.